src/main/java/org/medicmobile/webapp/mobile/EmbeddedBrowserActivity.java

```diff
@@ -44,262 +47,290 @@
+                    case GRAB_MRDT_PHOTO:
+                        String js = mrdt.process(requestCode, resultCode, i);
+                        trace(this, "Execing JS: %s", js);
+                        evaluateJavascript(js);
+                        return;
+                    default:
+                        trace(this, "onActivityResult() :: no handling for requestCode=%s", requestCode);
+                }
+            } else {
+                String js = simprints.process(requestCode, i);
+                trace(this, "Execing JS: %s", js);
+                evaluateJavascript(js);
+            }
+        } catch (Exception ex) {
+            String action = i == null ? null : i.getAction();
+            warn(ex, "Problem handling intent %s (%s) with requestCode=%s & resultCode=%s", i, action, requestCode, resultCode);
+        }
+    }
+
+    //> ACCESSORS
+    SimprintsSupport getSimprintsSupport() {
+        return this.simprints;
+    }
+
+    MrdtSupport getMrdtSupport() {
+        return this.mrdt;
+    }
+
+    SmsSender getSmsSender() {
+        return this.smsSender;
+    }
+
+    //> PUBLIC API
+    public void evaluateJavascript(final String js) {
+        container.post(new Runnable() {
+            public void run() {
+                // `WebView.loadUrl()` seems to be significantly faster than
+                // `WebView.evaluateJavascript()` on Tecno Y4.  We may find
+                // confusing behaviour on Android 4.4+ when using `loadUrl()`
+                // to run JS, in which case we should switch to the second
+                // block.
+                // On switching to XWalkView, we assume the same applies.
+                if (true) { // NOPMD
+                    container.load("javascript:" + js, null);
+                } else {
+                    container.evaluateJavascript(js, IGNORE_RESULT);
+                }
+            }
+        });
+    }
+
+    public void errorToJsConsole(String message, Object... extras) {
+        jsConsole("error", message, extras);
+    }
+
+    public void logToJsConsole(String message, Object... extras) {
+        jsConsole("log", message, extras);
+    }
+
+    //> PRIVATE HELPERS
+    private void jsConsole(String type, String message, Object... extras) {
+        String formatted = String.format(message, extras);
+        String escaped = formatted.replace("'", "\\'");
+        evaluateJavascript("console." + type + "('" + escaped + "');");
+    }
+
+    private void configureUseragent() {
+        String current = container.getUserAgentString();
+
+        container.setUserAgentString(createUseragentFrom(current));
+    }
+
+    private void openSettings() {
+        startActivity(new Intent(this,
+                SettingsDialogActivity.class));
+        finish();
+    }
+
+    private String getRootUrl() {
+        /*
+        * Removed Unnecessary Check as the Condition is always returning False*/
+        return appUrl.concat("/medic/_design/medic/_rewrite/");
```

Comment:
I think we actually want this to just be...
```suggestion
        return appUrl;
```

The "/medic/_design/medic/_rewrite/" bit is the old format that we no longer need.

Reply:
Removed concatinated string